Burst Headers Issue Fix

rcaseymsft opened the pull request:
# Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

<!-- Use the check list below to ensure your branch is ready for PR. -->

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code follows the code style of this project.
- [ ] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR
<!-- Concise description of the problem and the solution or the feature being added -->
To fix a burst issue that was preventing x-load-feedback header from being injected

## Does this introduce a breaking change

- [ ] YES
- [x] NO

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->
## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/1076

kforeverisback reviewed (approved):
⚡ LGTM
_[Optional Request]_ @gortegaMS  @rcaseymsft Is it possible to remove the following portion (namespace creation) from `burst-metrics-service.yaml` file? I think this portion is redundant and is different from `namespace.yaml` file.

```yaml
# Line 4~7
kind: Namespace
apiVersion: v1
metadata:
  name: burstservice
---
```

AAkindele commented on deploy/dev-ngsa-asb-westus2/burst/burst-flux-kustomization.yaml:
```diff
@@ -10,7 +10,7 @@ spec:
   interval: 1m0s # detect drift and undo kubectl edits every minute
   prune: true # remove stale resources from cluster
-  targetNamespace: burstservice # overwrite the ns for resources in path
+  # targetNamespace: burstservice # overwrite the ns for resources in path
```
Out of curiosity, how did this fix the issue?

kforeverisback replied:
@AAkindele 
Burstmetrics service is deployed in `burstservice` namespace. But in burst directory there is an [envoy filter file](https://github.com/retaildevcrews/ngsa-asb/blob/main/deploy/dev-ngsa-asb-westus2/burst/remote-filter-ngsa-cosmos.yaml) which needed to be applied to `ngsa` namespace (for istio-sidecar-proxy).

Now, with `targetNamespace:` flux overrides the namespace component of each file, under this `flux-kustomization`. Thus, the `envoyfilter` wouldn't get deployed in `ngsa` namespace.
No envoyfilter == no burstheader

rcaseymsft commented:
> ⚡ LGTM _[Optional Request]_ @gortegaMS @rcaseymsft Is it possible to remove the following portion (namespace creation) from `burst-metrics-service.yaml` file? I think this portion is redundant and is different from `namespace.yaml` file.
> 
> ```yaml
> # Line 4~7
> kind: Namespace
> apiVersion: v1
> metadata:
>   name: burstservice
> ---
> ```

We incorporated this change into the working wcnp-dev branch that's being used for the migration to the new subscription, so that issue will be addressed when the branch is merged into main. This PR serves the purpose of patching the issue temporarily until the migration is finalized.

I will create a triage item addressing your comment, mainly to preserve it and ensure that it's addressed in the PR review for wcnp-dev.
